lib/csv/csv.nit

```diff
@@ -167,79 +200,52 @@ class CsvWriter
 	# Note: Escaping may not be correctly recognized by some parsers.
 	var always_escape = false is writable
 
-	# Create a new writer with the specified format.
-	init with_format(ostream:Writer, format: CsvFormat) do
-		init(ostream)
-		self.format = format
-	end
-
-	# Append the specified sequence as a row.
-	#
-	# The representation of each cell is determined by `to_s`.
-	fun write_sequence(row: SequenceRead[Object]) do
-		if not row.is_empty then
-			var i = row.iterator
-			var separator = format.separator.to_s
-			write_cell i.item.to_s
-			i.next
-			for cell in i do
-				ostream.write separator
-				write_cell cell.to_s
-			end
-		end
-		ostream.write format.eol
-	end
+	# Write several lines to a stream
+	fun write_lines(lines: Array[Array[Object]]) do for i in lines do write_line i
 
 	# Append the specified row.
 	#
 	# The representation of each cell is determined by `to_s`.
-	fun write_row(row: Object...) do write_sequence(row)
-
-	# Close the output stream.
-	fun close do ostream.close
-
-	private fun write_cell(cell: String) do
-		if cell.is_empty then return
-		if not always_escape and format.is_value_clean(cell) then
-			ostream.write cell
-		else
-			ostream.write format.escape_cell(cell)
+	fun write_line(line: Array[Object]) do
```

Comment:
Same concept, I just could not get my head to use the `record` internally, I'll uniformize it in the doc